app/src/main/java/com/github/studydistractor/sdp/distraction/DistractionService.kt

```diff
@@ -0,0 +1,16 @@
+package com.github.studydistractor.sdp.distraction
+
+import androidx.compose.runtime.snapshots.SnapshotStateList
+interface DistractionService {
+
+    /**
+     * Fetch the distractions from the database
+     */
+    fun fetchDistractions() : SnapshotStateList<Distraction>

```

Comment:
Why a SnapshotStateList instead of a regular MutableStateList? Also maybe [LazyColumn](https://developer.android.com/jetpack/compose/lists?hl=fr) are the way to go

Reply:
Because the signature is not supported (I don't know why) and LazyColumns are used in screens